### PR TITLE
OAuth: Removes send_client_credentials_via_post setting 

### PR DESCRIPTION
**Pull request by LK4D4**
**What this PR does / why we need it**:
Fixes old staticcheck issues

**Which issue(s) this PR fixes**:
Not entirely but #8968

### CODE REVIEW

File: pkg/login/social/social.go [hunk 1]
```diff
@@ -65,36 +65,31 @@ func NewOAuthService() {
...
 		if !info.Enabled {
 			continue
 		}
 
-		// handle the clients that do not properly support Basic auth headers and require passing client_id/client_secret via POST payload
-		if info.SendClientCredentialsViaPost {
-			// TODO: Fix the staticcheck error
-			oauth2.RegisterBrokenAuthHeaderProvider(info.TokenUrl) //nolint:staticcheck
-		}
-
```
Comment (torkelo) [inline pkg/login/social/social.go]: why is this removed?

File: pkg/login/social/social.go [hunk 2]
```diff
@@ -65,36 +65,31 @@ func NewOAuthService() {
...
+			EmailAttributePath: sec.Key("email_attribute_path").String(),
+			AllowedDomains:     util.SplitString(sec.Key("allowed_domains").String()),
+			HostedDomain:       sec.Key("hosted_domain").String(),
+			AllowSignup:        sec.Key("allow_sign_up").MustBool(),
+			Name:               sec.Key("name").MustString(name),
+			TlsClientCert:      sec.Key("tls_client_cert").String(),
+			TlsClientKey:       sec.Key("tls_client_key").String(),
+			TlsClientCa:        sec.Key("tls_client_ca").String(),
+			TlsSkipVerify:      sec.Key("tls_skip_verify_insecure").MustBool(),
+			// TODO: Deprecate send_client_credentials_via_post as auth style is autodetected
```
Comment (torkelo) [inline pkg/login/social/social.go]: what was changed?

Comment (LK4D4) [reply pkg/login/social/social.go]: I added comment and gofmt reformated structure initialization

File: pkg/login/social/social.go (hunk 1 above)
Comment (LK4D4) [reply pkg/login/social/social.go]: This method was deprecated and does nothing.
https://godoc.org/golang.org/x/oauth2#RegisterBrokenAuthHeaderProvider

**Review by marcusolsson — APPROVED** [review]
LGTM

**Review by aknuds1 — APPROVED** [review]
LGTM

**Review by marefr — CHANGES_REQUESTED** [review]
Should add deprecation note for this section in docs: https://github.com/grafana/grafana/blob/master/docs/sources/auth/generic-oauth.md#set-up-oauth2-with-non-compliant-providers

Add deprecation notes:
https://github.com/grafana/grafana/blob/master/conf/defaults.ini#L383
https://github.com/grafana/grafana/blob/master/conf/sample.ini#L340

**Review by marefr — CHANGES_REQUESTED** [review]
Should add deprecation note for this section in docs: https://github.com/grafana/grafana/blob/master/docs/sources/auth/generic-oauth.md#set-up-oauth2-with-non-compliant-providers

Add deprecation notes:
https://github.com/grafana/grafana/blob/master/conf/defaults.ini#L383
https://github.com/grafana/grafana/blob/master/conf/sample.ini#L340

**Comment by marefr** [comment]
Will push some changes

File: docs/sources/installation/upgrading.md [hunk 3]
```diff
@@ -222,3 +222,7 @@ Pre Grafana 6.5.0, the CloudWatch datasource used the GetMetricStatistics API fo
...
 
 The handling of multi-valued template variables in dimension values has been changed in Grafana 6.5. When a multi template variable is being used, Grafana will generate a search expression. In the GetMetricData API, expressions are limited to 1024 characters, so it might be the case that this limit is reached when a multi-valued template variable that has a lot of values is being used. If this is the case, we suggest you start using `*` wildcard as dimension value instead of a multi-valued template variable.
+
+## Upgrading to v6.6
+
+The Generic OAuth setting, `send_client_credentials_via_post`, used for supporting non-compliant providers have been removed. From now on Grafana will automatically detect if credentials should be sent as part of url or body for a specific provider. The result will be remembered and used for additional OAuth requests for that provider.
```
Comment (aknuds1) [inline docs/sources/installation/upgrading.md]:
```suggestion
The Generic OAuth setting `send_client_credentials_via_post`, used for supporting non-compliant providers, has been removed. From now on, Grafana will automatically detect if credentials should be sent as part of the URL or request body for a specific provider. The result will be remembered and used for additional OAuth requests for that provider.
```

File: docs/sources/installation/upgrading.md [hunk 4]
```diff
@@ -85,7 +85,7 @@ sudo apt-get update
 
 #### Upgrade from binary .tar file
 
-If you downloaded the binary `.tar.gz` package, then you can just download and extract the new package and overwrite all your existing files. However, this might overwrite your config changes. 
+If you downloaded the binary `.tar.gz` package, then you can just download and extract the new package and overwrite all your existing files. However, this might overwrite your config changes.
```
Comment (aknuds1) [inline docs/sources/installation/upgrading.md]:
```suggestion
If you downloaded the binary `.tar.gz` package, you can just download and extract the new package and overwrite all your existing files. However, this might overwrite your config changes.
```

File: docs/sources/installation/upgrading.md [hunk 5]
```diff
@@ -85,7 +85,7 @@ sudo apt-get update
 
 #### Upgrade from binary .tar file
```
Comment (aknuds1) [inline docs/sources/installation/upgrading.md]:
```suggestion
#### Upgrade from binary .tar.gz file
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
I suggested some doc improvements.

File: docs/sources/installation/upgrading.md (hunk 4 above)
Comment (marefr) [reply docs/sources/installation/upgrading.md]: These are unrelated so suggest adding them in separate PR

File: docs/sources/installation/upgrading.md (hunk 5 above)
Comment (marefr) [reply docs/sources/installation/upgrading.md]: These are unrelated so suggest adding them in separate PR

File: docs/sources/installation/upgrading.md (hunk 4 above)
Comment (aknuds1) [reply docs/sources/installation/upgrading.md]: OK @marefr 👍

**Review by aknuds1 — APPROVED** [review]
LGTM

**Comment by oddlittlebird** [comment]
Doc changes look good.

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
@LK4D4 Thank you for contributing to Grafana!

**Comment by LK4D4** [comment]
Thanks for figuring that out :) Sorry I didn't have time to do it myself.
